### ISSUE TITLE
SUPERB BNECHMARK. 

### ISSUE THREAD

**Issue by raotnameh** [s3prl/s3prl#360]
Hi, 

I have a quick question. To simulate the same training and testing settings for the SUPERB benchmark should we just run the commands as provided in the README? Or changing the batch size or learning rate is not a bad habit for benchmarking.

**Comment by leo19941227** [comment]
In the SUPERB we change learning rate to explore the best one. So it is nice to explore the LR.
However, as for batch size, I do not recommended to change it. Since for some tasks (like ASV) it really get better results for larger batch size. However, it is not a good thing that a representation is SOTA just because it is benchmarked with more GPUs in my point of view. It is already not affordable for people not in industry to pre-train, and it might not be good that even for benchmarking people in school can't afford it. So I would suggest to use the pre-defined batch size (which fit in a single 16GB GPUs) to have a more fair comparison.

**Comment by raotnameh** [comment]
@leo19941227 Thank you for the response. Quick question: Can you also share the LR explorations used in your default settings?

**Comment by leo19941227** [comment]
Hi, it would be time-consuming for me to list all of them, but for the most common case wav2vec2/hubert base and large, we have released the training artifacts with their LR here

http://140.112.21.28:8000/

**Comment by raotnameh** [comment]
Not able to open the link.

**Comment by leo19941227** [comment]
Hmmmm.... But I can access it with my internet outside my lab.
Could you show your webpage? Is it blocked by being non-https, or it is completely not accessible.

**Comment by raotnameh** [comment]
![web](https://user-images.githubusercontent.com/46722949/173567070-984f58c5-776c-4487-a809-9cf07083a550.PNG)

**Comment by raotnameh** [comment]
Yesterday I was trying to download the data using http://140.112.21.28:9000/fluent.tar.gz. The same situation I was in.

**Comment by andi611** [comment]
> Not able to open the link. I think the ip addr is not public.

Strange, I'm able to open the link on my mobile network:
<img width="480" alt="image" src="https://user-images.githubusercontent.com/25531495/173571619-a9342069-d798-426e-ad50-ff80538c4e0d.png">

@leo19941227 maybe it is restricting connections from outside our country?

**Comment by raotnameh** [comment]
Strange. I just switched from Microsoft wifi to my phone network and it opens. Sorry for reporting the link not working.

**Comment by leo19941227** [comment]
I guess it might be that the network administrative in Microsoft automatically block any non-https access since it is not safe. Glad it is accessible.

**Comment by raotnameh** [comment]
@leo19941227 Can you share what other tasks' performance increases with a bigger batch size? 

"Since for some tasks (like ASV) it really get better results for larger batch size."

**Comment by raotnameh** [comment]
@leo19941227 I have a quick question regarding the num_Steps. 

For the KS task, why do we need **200000** steps? I mean, the model converges in very few epochs. 

The same goes for the tasks which are just "mean-pooling followed by a linear transformation with cross-entropy loss for utterance level tasks (KS, SID, IC, and ER). " 

For the ER task, the num_steps are just **30000**.

**Comment by raotnameh** [comment]
@leo19941227 I wanted to ask that Wavlm uses abusurd big batches for SUPERB benchmark. I tried Wavlm for the SID task with the 32 batch size and the results are somewhat similar to the HuBERT. Any take on this?

**Comment by leo19941227** [comment]
Hi @raotnameh ,

Thanks for reporting. This is indeed an issue.
I will consider to rerun the WavLM numbers for an upcoming journal.

**Comment by raotnameh** [comment]
@leo19941227 FYI: In their own paper the author show similar SID performance with (little better) and without (little worse) the denoising task. 
<img width="1349" alt="Screenshot 2023-12-14 at 9 51 01 PM" src="https://github.com/s3prl/s3prl/assets/46722949/dee851f3-bfa9-47b4-a653-74829089081e">

**Comment by raotnameh** [comment]
@leo19941227 Any updates on this? Could you please just run the 3 speaker-based tasks i.e., SID, ASV and SD for the time being and share the results with the community?

**Comment by leo19941227** [comment]
Hi @raotnameh ,

Unfortunately, currently I do not have enough resource to study the hypers for tuning WavLM. So I guess I can only leave it alone for now.
For your second question, what is the motivation to train 3 classifiers? Like, if you get improvement with training 3 classifiers, whether the other upstream can also get improvement with this modification?

**Comment by raotnameh** [comment]
Let's assume, we propose an architecture change which produces two pieces of information after each layer which can be used for a simple classification task. One information piece is similar to the HuBERT or WavLM. 

Give this information, we can train 3 classifiers using the proposed architecture. 
1. Using information 1, similar to the HuBERT or WavLM.
2. Using information 2 
3. And a combination of 1 and 2. 

What I proposed was to train these 3 classifiers on the above 3 combinations and then take a majority voting to get the final answer. 

Therefore to answer your question: "whether the other upstream can also get improvement with this modification?"
I do not know, because in theory we are not making changes which are applicable to other upstream models and are specific to our proposed model.

**Comment by leo19941227** [comment]
Hi,

Thanks for the information! If I am the reviewer, I would want to know why not simply use the 3rd evaluation result as the final result, which is more intuitive and more aligned with the original protocol of SUPERB. The ensemble process seems complicated and more like a performance boosting trick instead of really evaluating the performance of the representation.

P.s. I would like to clarify that my opinion does not represent any sort of "SUPERB official response" since I can not represent SUPERB on my own. Second, even if there is any "SUPERB official response", eventually what really matters is whether the argument can convince others instead of any official certification.

**Comment by raotnameh** [comment]
Ensemble is simply training 3 classifiers, not a complicated process. 

Anyways, Thank you for your point of view. Appreciate it.
